### DISCUSSION TITLE
Could `mrgsim_q()` have the `obsonly` option?

### DISCUSSION THREAD

**Discussion by benjaminrich** [metrumresearchgroup/mrgsolve#938] [Q&A] [answered]
It would be nice to be able to use `obsonly=TRUE` with `mrgsim_q()`.

**Comment by kylebaron** [comment, marked as answer]
Hi @benjaminrich - 

Yeah, I can add that. Can you try `qsim()` in the meantime? It has that argument. I had been developing those functions in parallel trying to reduce overhead on very small / quick turnaround problems. I think they would be hard to differentiate right now. 

Also - would you mind giving me a (ballpark) idea how you're using it and how much benefit you think you're getting? Just a guestimate  ... "seems like it's helping" or "can't tell, but using it anyway" etc. 

Thanks,
Kyle

**Comment by benjaminrich** [comment]
I see. I was not aware of `qsim()` but indeed it looks like it would suit my purpose. Thanks.

As to the increase in speed, I cannot give even a ballpark figure right now unfortunately. But give me a bit of time and I will try to.

**Comment by kylebaron** [comment]
Ok; no problem. Sometimes I just use it too without regard to any savings; just curious about it. 

If you are using `v1.0.0` that does include some refactor of the standard `mrgsim()` simulation path that also closes the gap; I was sort of satisfied that the difference was fairly small and could only be detectable for very quick simulation that was repeated a very large number of times. So if you need the other features, I'd switch to that interface, esp if you don't know there is a benefit. 

Kyle
